pkg/odo/cli/url/create.go

```diff
@@ -77,14 +79,22 @@ func (o *URLCreateOptions) Validate() (err error) {
 // Run contains the logic for the odo url create command
 func (o *URLCreateOptions) Run() (err error) {
 
-	log.Infof("Adding URL to component: %v", o.Component())
 	urlRoute, err := url.Create(o.Client, o.urlName, o.componentPort, o.Component(), o.Application)
 	if err != nil {
 		return err
 	}
 	urlCreated := url.GetURLString(urlRoute.Spec.Protocol, urlRoute.Spec.URL)
-	log.Successf("URL created for component: %v\n\n"+
-		"%v - %v\n", o.Component(), urlRoute.Name, urlCreated)
+
+	if o.outputFlag == "json" {
```

Comment:
see my comment https://github.com/redhat-developer/odo/pull/1404/files#r261588399

This if is common for all outputs, we need to abstract this away, so we can easily add more formats in the future.